clients/rust/src/lib.rs

```diff
@@ -160,13 +162,25 @@ impl ClientBuilder {
     /// Builds a `Client` in HTTPGateway mode, erroring if the mode is not HTTPGateway
     /// This allows avoiding calling the async `build` method
     pub fn build_http(self) -> Result<Client, ClientBuilderError> {
-        let ClientBuilderMode::HTTPGateway { url } = self.mode else {
+        let ClientBuilderMode::HTTPGateway { url, timeout } = self.mode else {
             return Err(ClientBuilderError::NotHTTPGateway);
         };
+        let http_client = match self.http_client {
+            Some(http_client) => http_client,
```

Comment:
This seems like a footgun - I think we should either error if a client is already provided, or remove `timeout` from `HTTPGateway` and require the caller to pass in a customized `Client` with the desired timeout set.